src/contact.rs

```diff
@@ -1443,6 +1445,61 @@ fn split_address_book(book: &str) -> Vec<(&str, &str)> {
         .collect()
 }
 
+#[derive(Debug)]
+pub(crate) struct RecentlySeenLoop {
+    /// Task running "recently seen" loop.
+    handle: task::JoinHandle<()>,
+
+    interrupt_send: Sender<()>,
+}
+
+impl RecentlySeenLoop {
+    pub(crate) fn new(context: Context) -> Self {
+        let (interrupt_send, interrupt_recv) = channel::bounded(1);
+
+        let handle = task::spawn(async move { Self::run(context, interrupt_recv).await });
+        Self {
+            handle,
+            interrupt_send,
+        }
+    }
+
+    async fn run(context: Context, interrupt: Receiver<()>) {
+        loop {
+            let now = time();
+            if let Some((contact_id, next_unseen)) = context
+                .sql
+                .query_row_optional(
+                    "SELECT MIN(last_seen) FROM contacts WHERE last_seen > ?",
+                    paramsv![now + SEEN_RECENTLY_SECONDS],
+                    |row| {
+                        let contact_id: ContactId = row.get(0)?;
+                        let timestamp: u64 = row.get(1)?;
+                        Ok((contact_id, timestamp))
+                    },
+                )
+                .await
+                .unwrap_or_default()
+            {
+                // TODO timeout instead of sleep
+                tokio::time::sleep(std::time::Duration::from_secs(next_unseen)).await;
+                context.emit_event(EventType::ContactsChanged(Some(contact_id)))
+            } else {
+                // TODO wait for interrupt
+                tokio::time::sleep(std::time::Duration::from_secs(60)).await;
+            }
+        }
```

Comment:
How is the interruption actually working? I don't see interrupt_send being used here?